src/redux/missions/missionsSlice.js

```diff
@@ -14,7 +14,20 @@ export const getMissions = createAsyncThunk('missions', async () => {
 const missionsSlice = createSlice({
   name: 'rockets',
   initialState,
-  reducers: {},
+  reducers: {
+    joinMission: (state, action) => {
+      const reserved = [];
+      state.missions.forEach((mission) => {
+        if (mission.mission_id === action.payload.id) {
+          const reservedMission = { ...mission, reserved: !mission.reserved };
+          reserved.push(reservedMission);
+        } else {
+          reserved.push(mission);
+        }
+      });
```

Comment:
Great work on implementing join mission reducer, however, kindly avoid state mutation. I would suggest you use `state.missions.map()` to avoid directly mutating the state. It's redux best practice and also a project requirement.